src/lib.js

```diff
@@ -193,31 +193,43 @@ var curlConverter = {
         return arr.join("&");
     },
 
+    sanitizeCurlArgs: function(curlString) {
+        //replace -XPOST with -X POST
+        curlString = curlString.replace(/(-X)([A-Z]+)/, function (match, x, method) {return x + " " + method;})
+
+        var argv = shellQuote.parse("node " + curlString, function(key) {
+            // this is done to prevent converting vars like $id in the curl input to ''
+            return '$' + key;
+        });
+        var sanitizedArgs = _.map(_.filter(argv, function(arg) { return !_.isEmpty(arg) }), function (arg) {
+          if (_.isObject(arg) && arg.op === 'glob') {
+            return arg.pattern
+          }
+          else if (arg.op && arg.op.startsWith('$') && arg.op.length > 3) {
+            // in the app, certain headers like -H $'cookie: abc' are treated as operators
+            // converting the arg to cookie: abc instead of op: $'cookie: abc'
+            return arg.op.substring(2, arg.op.length-1);
+          }
+          else if (arg.startsWith('$') && arg.length > 2) {
```

Comment:
will `-X` be part of `arg`?

Reply:
@Dhroov7 ^
This is for the `converting the arg to -X 'POST'` comment.

Reply:
also add a link to the spec/RFC which talks about what $'POST' is supposed to mean

Reply:
@abhijitkane Okay!!

Reply:
@Dhroov7 There are 2 comments before the RFC one too.

Reply:
@abhijitkane Yes, `-X` will be part of `arg`.